ksql-engine/src/main/java/io/confluent/ksql/ddl/commands/DropSourceCommand.java

```diff
@@ -53,11 +54,23 @@ public DDLCommandResult run(MetaStore metaStore) {
           dataSourceType == DataSource.DataSourceType.KSTREAM ? "STREAM" : "TABLE"
       ));
     }
-    DropTopicCommand dropTopicCommand =
-        new DropTopicCommand(dataSource.getKsqlTopic().getTopicName());
+    if (!referentialIntegrityTable.isSafeToDrop(sourceName)) {
```

Comment:
Won't this always be false for sources created by CSAS/CTAS statements, since those sources have their creating query in the reference table?